### PR TITLE
[BottomAppBar] Examples use semantic color scheme.

### PR DESCRIPTION
**Pull request by romoore**
The BottomAppBar now supports the "surface variant" of the Material
theming color themers. The examples should be updated to reflect the
correct styling for Material Theming.

**This is a breaking change that will break internal MDC Catalog screenshot tests.**

|Example|Before|After|
|--|--|--|
|Catalog (Objc)|bab-objc-catalog-before|bab-objc-catalog-after|
|Dragons (Objc)|bab-objc-dragon-before|bab-objc-dragon-after|
|Dragons (Swift)|bab-swift-dragon-before|bab-swift-dragon-after|

Part of #3928

### CODE REVIEW

File: components/BottomAppBar/examples/BottomAppBarTypicalUseExample.m [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #import <UIKit/UIKit.h>
 
 #import "MaterialBottomAppBar.h"
```
[the remark this answers is not in the document]
Comment (romoore) [reply components/BottomAppBar/examples/BottomAppBarTypicalUseExample.m]: This is a weird

[the remark this answers is not in the document]
Comment (romoore) [reply components/BottomAppBar/examples/BottomAppBarTypicalUseExample.m]: ...pair of comments.

**Review by jverkoey — APPROVED** [review]
LGTM but please remove the `!` breaking change annotation - we only consider src-breaking changes as breaking (this PR only includes changes to the examples).

**Pull request merged**